server/route.go

```diff
@@ -366,6 +368,8 @@ func (c *client) sendRouteConnect(tlsRequired bool) {
 		TLS:      tlsRequired,
 		Name:     c.srv.info.ID,
 		Headers:  c.srv.supportsHeaders(),
+		Cluster:  c.srv.ClusterName(),
+		Dynamic:  c.srv.getOpts().Cluster.Name == "",
```

Comment:
I wonder if we should not store the "dynamic" nature of the cluster name somewhere a bit more accessible. Also, when a server accepts the remote cluster name, and sets its own to that value, then I would think that this server's cluster name is no longer "dynamic".

Reply:
I still think its dynamic if the other side was dynamic, but if not agree I can clear it. Will look into that.

Reply:
I clear it now.